File: Crio/crio_programming_interview_problems-master/FirstUniqueInteger/FirstUniqueInteger.cpp
#include<bits/stdc++.h>
using namespace std;
int firstUniqueInteger(vector<int> &v) {
	//CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
	int n = v.size();
	queue<int> Q;
	map<int, int > M;
	for ( int i = 0; i < n; i++) {
		if (M.find(v[i]) == M.end()) {
			Q.push(v[i]);
		}
		M[v[i]]++;
	}
	while (!Q.empty()) {
		int number = Q.front();
		Q.pop();
		if (M[number] == 1) {
			return number;
		}
	}
	return -1;
	//CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
}
int main() {
	int n, m;
	cin >> n;
	vector<int> v(n);
	for ( int i = 0; i < n; i++) {
		cin >> v[i];
	}
	int ans = firstUniqueInteger(v);
	cout << ans << endl;
}

File: Crio/StackAndQueue/9_FirstUniqueInteger/FirstUniqueInteger.py
import queue

def firstUniqueInteger(lis):
    # CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    n = len(lis)
    q = queue.Queue()
    m = dict()
    for el in lis:
        if el not in m:
            q.put(el)
            m[el]=0
        m[el]+=1
    while not q.empty():
        number = q.queue[0]
        q.get()
        if m[number] == 1:
            return number
    return -1 
    # CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def main():
    n = int(input())
    lis = list(map(int,input().split()))
    assert len(lis) == n
    result = firstUniqueInteger(lis)
    print(result)

if __name__ == "__main__":
    main()
